src/exception/repo/entity.ts

```diff
@@ -101,16 +115,30 @@ const entityExceptionRepository: EntityExceptionRepository = {
     }
   },
 
-  async deleteSingleEntity(programId: string, entity: Entity) {
+  async deleteSingleEntity(programId: string, entity: Entity, submitterDonorIds: string[]) {
     L.debug(
       `deleting single entity ${entity} exception with program id: ${JSON.stringify(programId)}`,
     );
     try {
-      const update = { $set: { [entity]: [] } };
-      const doc = await EntityExceptionModel.findOneAndUpdate({ programId }, update, {
-        new: true,
-      }).lean(true);
-      return checkDoc(doc);
+      const entityExceptionDoc = await EntityExceptionModel.findOne({ programId });
+      if (entityExceptionDoc) {
+        /**
+         * typescript union array methods don't work well particulary pre v4 (currently on 3.9.5)
+         * all our entity types union with BaseEntityExceptionRecord
+         * filter only uses the `submitter_donor_id` field which is in BaseEntityExceptionRecord
+         * explicitly adding `any` typings so it's very obvious we loose type data here
+         */
+        const entitiesToFilter: any = entityExceptionDoc[entity];
```

Comment:
union array types don't use array methods well with 3.9.5
tried the various ways to get it to work, but no success.
likely will work better with upgraded typescript as a quick try in the TS playground was successful

we can at least use the `BaseExceptionRecord` so the filter function , the most important part, has some type info.

explicit `any` typing the rest so it's clear as day